packages/orm/src/EntityManager.ts

```diff
@@ -612,67 +630,62 @@ export class EntityManager<C = {}> {
     // But since it's a public API, we should just early exit.
+    const list = toArray(entityOrList).filter((e) => e !== undefined && (e.isPendingDelete || !e.isDeletedEntity));
     if (list.length === 0) {
       return !fn ? (entityOrList as any) : fn(entityOrList as any);
     }
 
     // If a bunch of `.load`s get called in parallel for the same entity type + load hint, dedup them down
     // to a single promise to avoid making more and more promises with each level/fan-out of a nested load hint.
-    const key = `${list[0]?.__orm.metadata.tagName}-${JSON.stringify(hint)}-${opts.forceReload}`;
+    const key = `${list[0]?.__orm.metadata.tagName}-${JSON.stringify(hintOpt)}-${opts.forceReload}`;
     const loader = getOrSet(
       this.populateLoaders,
       key,
       new DataLoader(
         (list) => {
-          // this.populates[key] = (this.populates[key] ?? 0) + 1;
-          const promises = list
-            .filter((e) => e !== undefined && (e.isPendingDelete || !e.isDeletedEntity))
-            .flatMap((entity) => {
-              // This implementation is pretty simple b/c we just loop over the hint (which is a key / array of keys /
-              // hash of keys) and call `.load()` on the corresponding o2m/m2o/m2m reference/collection object. This
-              // will kick in the dataloader auto-batching and end up being smartly populated (granted via 1 query per
-              // entity type per "level" of resolution, instead of 1 single giant SQL query that inner joins everything
-              // in).
-              if (typeof hint === "string") {
-                return (entity as any)[hint].load(opts);
-              } else if (Array.isArray(hint)) {
-                return (hint as string[]).map((key) => (entity as any)[key].load(opts));
-              } else if (typeof hint === "object") {
-                return Object.entries(hint as object).map(([key, nestedHint]) => {
-                  const relation = (entity as any)[key];
-                  return relation.load(opts).then((result: any) => {
-                    if (
-                      Object.keys(nestedHint).length > 0 &&
-                      ((result instanceof Array && result.length > 0) || result !== undefined)
-                    ) {
-                      return this.populate(result, { hint: nestedHint, ...opts });
-                    }
-                    return result;
-                  });
-                });
-              } else {
-                throw new Error(`Unexpected hint ${hint}`);
-              }
+          const hints = Object.entries(normalizeHint(hintOpt as any));
+
+          // One breadth-width pass to ensure each relation is loaded
+          const loadPromises = list.flatMap((entity) => {
+            return hints.map(([key]) => {
+              const relation = (entity as any)[key];
+              return relation.isLoaded && !opts.forceReload ? undefined : (relation.load(opts) as Promise<any>);
```

Comment:
This `relation.isLoaded` check is what surfaced the clone bug in `ManyToOneReference`.